### PR TITLE
feat(dashboard): Log in, reset password, and accept invite pages

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Adds the following pages to the new admin: /login, /reset-password, /reset-password/:token, and /invite.
- Adds `<Alert />` component to `@medusajs/ui`

### CODE REVIEW

**Comment by srindom** [comment]
CleanShot 2024-02-03 at 11  44 42@2x

I tried opening the invite link with an invalid token - maybe we should have an error message that the invite is invalid or something like that?

**Comment by srindom** [comment]
praise: love the animation when you sign up 😍

File: packages/admin-next/dashboard/src/routes/invite/invite.tsx [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+  const decodedToken: {
+    invite_id: string
+    role: UserRoles
+    user_email: string
+    iat: number
+  } | null = token ? decodeToken(token) : null
+
+  if (!decodedToken) {
+    tokenStatus = "invalid"
+  }
```
Comment (srindom) [inline packages/admin-next/dashboard/src/routes/invite/invite.tsx]: Have you tested this with a token that is not a JWT token - e.g. some random characters. Doesn't seem to result in an invalid token status

**Review by srindom — COMMENTED** [review]
Looks great - only issue I came across was the token status thing

File: packages/admin-next/dashboard/src/routes/invite/invite.tsx (hunk 1 above)
Comment (kasperkristensen) [reply packages/admin-next/dashboard/src/routes/invite/invite.tsx]: I did test it, and on my end it did fail when the token was not a valid JWT token, but I guess it was possible to have false positives, as long as decodeToken didn't return `null`. 

I have added a parse with zod, that validates that the decoded token has all the expected fields, e.g. a valid email, invite_id, etc, and if any one is missing then we show a invalid state screen instead 👍

Comment (srindom) [reply packages/admin-next/dashboard/src/routes/invite/invite.tsx]: Nice! LGTM 💪

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**